### PR TITLE
Fix rewards distribution amount decimals places

### PR DESCRIPTION
**Pull request by cris-santos**
### Summary

Make sure the rewards distribution amount decimals are 6 decimals.

### Details

Also increase the percentage decimals to 10.

---

- [x] **I, the PR author, declare that this PR works as expected and does not break any service. I also declare that I gave my best to apply all of the best practices and that I'm leaving the code better than I found it.**

## Release Notes

- **New Features**
	- Introduced a new function `stubBigNumber` for generating random BigNumber values with customizable maximum and decimal places.
	- Added a helper function `assertExpectedRewardsDistribution` to validate rewards distribution structures.
	- Added `splitBigNumberIntoParts` to split a BigNumber into specified parts.
	- Introduced a new test data file for rewards distribution testing.

- **Improvements**
	- Enhanced the test coverage for rewards distribution logic and mathematical helper functions.
	- Streamlined the structure and clarity of end-to-end tests for the rewards distribution processor.
	- Updated tests for rewards distribution to focus on relevant scenarios and improved error handling.

- **Refactor**
	- Restructured rewards distribution calculations and introduced new constants for decimal formatting.
	- Removed outdated test cases and refined the overall test structure for rewards distribution.
	- Updated types and constants to improve clarity and maintainability in the rewards distribution logic.

### CODE REVIEW

File: apps/methodologies/bold/rule-processors/credit/rewards-distribution/src/lib/rewards-distribution.helpers.spec.ts [hunk 1]
```diff
@@ -1,216 +1,171 @@
...
+        actorType: DocumentEventActorType.NETWORK,
+        participant: random<Participant>(),
+        percentage: '50',
+      };
+      const result = calculateRewardsDistribution(faker.number.float(), [
+        {
+          massValue: stubBigNumber(),
+          resultContent: {
+            certificateRewards: [actor, actor] as never,
+            massRewards: [] as never,
```
Comment (andtankian) [inline apps/methodologies/bold/rule-processors/credit/rewards-distribution/src/lib/rewards-distribution.helpers.spec.ts:102]: Nice!

**Review by andtankian — COMMENTED** [review]
LGTM.

However I personally miss some static assertions. It was awesome that you created dynamic assertion helpers but it seems a little more secure to have control of a scenario you know exactly the calculation result and assert the refactored code in that scenario, instead having it dynamically asserted.
The main problem here (imho) is that we just trust in your dynamic assertion helper and every test is passing as expected, but if the dynamic assertion helper was not implemented covering all the scenarios? I'm not saying the implementation is wrong however having static assert for known scenarios seems a little more secure.

For example, let's suppose we have a mass of value 1000.
Given 2 sources, 2 haulers, 2 networks, what would be the expected result?
In this case we have a known values of the reward distribution and we should check if the refactored code produces same known output. That's the point.

What do you think?

(No need for remove dynamic assertion helper but add at least one static assertion can make it little more secure)

**Comment by cris-santos** [comment]
> LGTM.
> 
> However I personally miss some static assertions. It was awesome that you created dynamic assertion helpers but it seems a little more secure to have control of a scenario you know exactly the calculation result and assert the refactored code in that scenario, instead having it dynamically asserted. The main problem here (imho) is that we just trust in your dynamic assertion helper and every test is passing as expected, but if the dynamic assertion helper was not implemented covering all the scenarios? I'm not saying the implementation is wrong however having static assert for known scenarios seems a little more secure.
> 
> For example, let's suppose we have a mass of value 1000. Given 2 sources, 2 haulers, 2 networks, what would be the expected result? In this case we have a known values of the reward distribution and we should check if the refactored code produces same known output. That's the point.
> 
> What do you think?

I think it's worth having both test cases, but between both of them, in this scenario, I think the dynamic one is preferred. When we test with only static data, we do not cover all of the scenarios, which is the issue that this PR is fixing. But I agree that it's worth it to test with static data. I'll put back the static test case.

**Review by AMarcosCastelo — DISMISSED** [review]

File: apps/methodologies/bold/rule-processors/credit/rewards-distribution/src/lib/rewards-distribution.test.data.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+              id: 'ttewegtjrygioqpnmjcak',
+              name: 'wupdjxggyfsxb',
+            },
+            percentage: '14',
+          },
+        ],
+        massRewards: [] as never,
+      },
+    },
+  ];
```
[the remark this answers is not in the document]
Comment (cris-santos) [reply apps/methodologies/bold/rule-processors/credit/rewards-distribution/src/lib/rewards-distribution.test.data.ts]: yes

**Review by AMarcosCastelo — APPROVED** [review]

**Pull request merged**